instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteAlternativesTest.kt

```diff
@@ -0,0 +1,81 @@
+package com.mapbox.navigation.instrumentation_tests.core
+
+import androidx.test.espresso.Espresso
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.navigation.base.options.NavigationOptions
+import com.mapbox.navigation.base.route.RouteAlternativesOptions
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.MapboxNavigationProvider
+import com.mapbox.navigation.instrumentation_tests.R
+import com.mapbox.navigation.instrumentation_tests.activity.EmptyTestActivity
+import com.mapbox.navigation.instrumentation_tests.utils.MapboxNavigationRule
+import com.mapbox.navigation.instrumentation_tests.utils.idling.RouteAlternativesIdlingResource
+import com.mapbox.navigation.instrumentation_tests.utils.location.MockLocationReplayerRule
+import com.mapbox.navigation.instrumentation_tests.utils.readRawFileText
+import com.mapbox.navigation.instrumentation_tests.utils.runOnMainSync
+import com.mapbox.navigation.testing.ui.BaseTest
+import com.mapbox.navigation.testing.ui.utils.getMapboxAccessTokenFromResources
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import java.util.concurrent.TimeUnit
+
+class RouteAlternativesTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.java) {
+
+    @get:Rule
+    val mapboxNavigationRule = MapboxNavigationRule()
+
+    @get:Rule
+    val mockLocationReplayerRule = MockLocationReplayerRule(mockLocationUpdatesRule)
+
+    private lateinit var mapboxNavigation: MapboxNavigation
+
+    @Before
+    fun setup() {
+        val routeAlternativesOptions = RouteAlternativesOptions.Builder()
+            .intervalMillis(TimeUnit.SECONDS.toMillis(10))
+            .build()
+        mapboxNavigation = MapboxNavigationProvider.create(
+            NavigationOptions.Builder(activity)
+                .accessToken(getMapboxAccessTokenFromResources(activity))
+                .routeAlternativesOptions(routeAlternativesOptions)
+                .build()
+        )
+    }
+
+    @Test
+    fun expect_faster_route_alternatives() {
```

Comment:
This test will try to make an actual route request to fetch the alternatives.

Could you add a mock web server setup as in other tests to isolate this variable?

Reply:
I did add a mock web server and it needed to be oddly specific to the implementation. Also I was having trouble getting it to work. It was working on my local machine, but wouldn't work on circle-ci. This was because my local machine had failing network requests, but had loaded the road network. So the onboard/offboard router causes the web server test only work for the offboard router. 

My conclusion was, our solution for alternative routes, should not be concerned with where the alternatives come from. Considering that this is an integration test, it seems appropriate to have it integrated with a real server.

Reply:
> Considering that this is an integration test, it seems appropriate to have it integrated with a real server.

This should be an integration test for the SDK, not for the Directions API. By pinging the real server we are exposing the test suite to the variability and instability that network connectivity or server latency/availability can introduce.

Let's figure out why you were hitting those issues and try to mock the responses correctly.

Reply:
When we use the mock server, the test becomes more complicated than network instability. This is because we then need to have pre-determined coordinates in which the next directions request will happen. Or we need to make a fake estimated route response. I found the deterministic test, only testing "if faster routes will make a network request" - which is [already tested by unit tests](https://github.com/mapbox/mapbox-navigation-android/pull/4375/files#diff-9d5de4542e0fc70f55cc5dc3007a2636f073fb9a262f37b906bfcae09b72140eR89)

I'm taking another look to see if I see a different way that could be good.

Reply:
Ok updated the test. We will need to regenerate the server responses any time we change test parameters, but it's all hooked up to a mock server.